src/components/DocumentTree/model.js

```diff
@@ -0,0 +1,115 @@
+import {
+    createDocument,
+    deleteDocument,
+} from "../../services/apiManager/index.js";
+import requireNew from "../../services/requireNew.js";
+
+import { initialDocument } from "../../constants/initialData.js";
+import { replaceDocumentId } from "../../services/router.js";
+
+const validateSelf = (data) => {
+    if (typeof data.id !== "number")
+        throw new Error("문서 id가 숫자 타입이 아닙니다");
+    else if (typeof data.title !== "string")
+        throw new Error("문서 title가 문자열 타입이 아닙니다");
+    else if (Array.isArray(data.documents) === false)
+        throw new Error("문서 documents가 배열이 아닙니다");
+};
+
+export const validateDocumentNode = (nodes) => {
+    if (Array.isArray(nodes) === false) {
+        throw new Error("documentTree의 nextState가 배열이 아닙니다");
+    }
+
+    try {
+        nodes.forEach((node) => validateSelf(node));
+    } catch (err) {
+        throw err;
+    }
+};
+
+export function DocumentNode({ initialData, appendNode, findRootOf }) {
+    try {
+        requireNew(new.target);
+        validateSelf(initialData);
+    } catch (err) {
+        console.error(err);
+        return;
+    }
+
+    const $documentNode = document.createElement("ul");
+    const $addChildButton = document.createElement("button");
+    const $deleteButton = document.createElement("button");
+    const $title = document.createElement("span");
+
+    $documentNode.appendChild($title);
+    $title.classList.add("document-title");
+
+    $documentNode.appendChild($addChildButton);
+    $addChildButton.classList.add("add-button");
+
+    $documentNode.appendChild($deleteButton);
+    $deleteButton.classList.add("delete-button");
+
+    $documentNode.classList.add("document-node");
+    
+    this.id = initialData.id;
+    this.title = initialData.title;
+    this.documents = initialData.documents;
```

Comment:
요 부분 아까 제안드렸던 dom 함수를 사용하면 이렇게 변경해볼 수 있어요!

```suggestion
 const $addChildButton = createCustomElement({
    tag: 'button',
    props: { className: 'add-button' },
  });
    
  const $deleteButton = createCustomElement({
    tag: 'button',
    props: { className: 'delete-button' },
  });
  
 const $title = createCustomElement({
    tag: 'span',
    props: { className: 'document-title' },
    children: [$headerImg, $headerTitle],
  });

 const $documentNode = createCustomElement({
    tag: 'ul',
    props: { className: 'document-node' },
    children: [$title, $addChildButton, $deleteButton],
  });
```
appendChild, classList 반복문을 함수를 통해 처리할 수 있습니다!

Reply:
아 진짜 훨~~~~~~~~~~~~~~씬 깔끔하네요!

이렇게 직접 타이핑까지 해주시니까 진짜 너무 감사합니다.. ㅠㅠ

props 에는 원하는 속성을 넣고

children으로 하위 요소를 넣고

관리가 훨씬 수월해지네요 ㄷㄷㄷ